From 73d52eb4c66e79e3163836f11b0d42c3c3b001bb Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:02:18 -0400
Subject: [usb] xhci: refactor TRB_DISABLE_SLOT case into function

Message-id: <1410887463-20674-67-git-send-email-dzickus@redhat.com>
Patchwork-id: 95283
O-Subject: [RHEL7 PATCH 066/591] xhci: refactor TRB_DISABLE_SLOT case into function
Bugzilla: 1032371
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1032371

commit 6c02dd147a7a3dbbfc7451abbf58b3b2a2a5483d
Author: Xenia Ragiadakou <burzalodowa@gmail.com>
Date:   Mon Sep 9 13:29:48 2013 +0300

    xhci: refactor TRB_DISABLE_SLOT case into function

    The function that handles xHCI command completion is much too long and
    there is need to be broken up into individual functions for each command
    completion to improve code readablity.
    This patch refactors the code in TRB_DISABLE_SLOT switch case in
    handle_cmd_completion() into a fuction named xhci_handle_cmd_disable_slot().

    Signed-off-by: Xenia Ragiadakou <burzalodowa@gmail.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci-ring.c b/drivers/usb/host/xhci-ring.c
index ea41118..46f1426 100644
--- a/drivers/usb/host/xhci-ring.c
+++ b/drivers/usb/host/xhci-ring.c
@@ -1394,6 +1394,19 @@ static void xhci_handle_cmd_enable_slot(struct xhci_hcd *xhci, int slot_id,
  complete(&xhci->addr_dev);
 }
 
+static void xhci_handle_cmd_disable_slot(struct xhci_hcd *xhci, int slot_id)
+{
+ struct xhci_virt_device *virt_dev;
+
+ virt_dev = xhci->devs[slot_id];
+ if (!virt_dev)
+  return;
+ if (xhci->quirks & XHCI_EP_LIMIT_QUIRK)
+  /* Delete default control endpoint resources */
+  xhci_free_device_endpoint_resources(xhci, virt_dev, true);
+ xhci_free_virt_device(xhci, slot_id);
+}
+
 static void handle_cmd_completion(struct xhci_hcd *xhci,
   struct xhci_event_cmd *event)
 {
@@ -1451,13 +1464,7 @@ static void handle_cmd_completion(struct xhci_hcd *xhci,
     GET_COMP_CODE(le32_to_cpu(event->status)));
   break;
  case TRB_TYPE(TRB_DISABLE_SLOT):
-  if (xhci->devs[slot_id]) {
-   if (xhci->quirks & XHCI_EP_LIMIT_QUIRK)
-    /* Delete default control endpoint resources */
-    xhci_free_device_endpoint_resources(xhci,
-      xhci->devs[slot_id], true);
-   xhci_free_virt_device(xhci, slot_id);
-  }
+  xhci_handle_cmd_disable_slot(xhci, slot_id);
   break;
  case TRB_TYPE(TRB_CONFIG_EP):
   virt_dev = xhci->devs[slot_id];
-- 
1.7.1